src/task_scl_eff_pot_hab.py

```diff
@@ -188,49 +165,72 @@ def density_to_patch_size(ft):
-        ).reproject(crs="EPSG:4326", scale=self.scale)
-
-        polygonReducer = (
-            ee.Reducer.mode().unweighted()
-            .combine(ee.Reducer.mode().unweighted(), "hii_")
-            .combine(ee.Reducer.sum().unweighted(), "area_")
-            .combine(ee.Reducer.mode().unweighted(), "patch_")
-            .combine(ee.Reducer.mode().unweighted(), "stp_stn_")
+
+        potential_core = (
+            ee.Image(0).where(potential_habitat.gte(min_patch_size), 1).selfMask()
         )
-        hii_grp_poly = hii_grp.reduceToVectors(
-            reducer=polygonReducer,
-            geometry=ee.Geometry.MultiPolygon(self.aoi).bounds(),
-            crs=self.crs,
-            scale=self.scale,
-            maxPixels=self.ee_max_pixels,
-            geometryInNativeProjection=True,
-        ).select(
-            ["area_sum", "hii_mode", "label", "mode", "patch_mode", "stp_stn_mode"],
-            [
-                "patch_area",
-                "hii",
-                "label",
-                "prob",
-                "min_patch_area",
-                "min_stp_stn_area",
-            ],
+
+        potential_stepping_stone = (
+            ee.Image(0)
+            .where(
+                potential_habitat.lt(min_patch_size).And(
+                    potential_habitat.gte(min_stepping_stone_size)
+                ),
+                1,
+            )
+            .selfMask()
         )
 
-        scl_eff_pot_hab = "{}/geographies/Sumatra/scl_poly/{}/scl_eff_pot_hab".format(
-            self.species, self.taskdate
+        potential_core = (
+            dilate(potential_core, 2)
+            .reproject(crs="EPSG:4326", scale=self.scale)
+            .multiply(3)
+            .unmask(0)
+            .updateMask(self.water)
         )
+        potential_stepping_stone = (
+            dilate(potential_stepping_stone, 2)
+            .reproject(crs="EPSG:4326", scale=self.scale)
+            .unmask(0)
+            .updateMask(self.water)
+        )
+
+        scl_polys = potential_core.add(potential_stepping_stone).selfMask()
+
+        # Temp geometry for RFE scenario
+        # geometry = ee.Geometry.Polygon(
+        #     [
+        #         [
+        #             [129.28241561583914, 51.490498638741165],
+        #             [129.28241561583914, 42.20681310459823],
+        #             [140.35663436583914, 42.20681310459823],
+        #             [140.35663436583914, 51.490498638741165],
+        #         ]
+        #     ],
+        #     None,
+        #     False,
+        # )
+
+        scl_polys = (
+            ee.Image(1)
+            .updateMask(scl_polys)
+            .addBands(scl_polys)
+            .addBands(self.probability)
+            .rename(["constant", "scl_poly", "probability"])
+            .reduceToVectors(
+                reducer=ee.Reducer.max(),
+                geometry=self.extent,
+                scale=self.scale,
+                crs="EPSG:4326",
+                maxPixels=1e10,
```

Comment:
`EETask` has a default
`ee_max_pixels = 500000000000`
which should be used for this arg. You can override it in the `__init__` or at top of class (i.e. with `scale`)

Reply:
good point. done.  I don't think there is a reason not to use the larger number. early on in earth engine developer interactions they made a point to say set this at a reasonable level.  I think it was to make sure people didn't blow up the resources